### PR TITLE
Limit UI

### PR DESCRIPTION
**Pull request by guergana**
number validation to be done in another PR. This is already too big.

Commits 2-4 were part of [PR 119](https://github.com/wmde/query-builder/pull/119) and have been merged into this PR and were already reviewed by @micgro42

### CODE REVIEW

File: public/i18n/en.json [hunk 1]
```diff
@@ -17,5 +17,6 @@
   "query-builder-add-condition": "Add condition",
   "query-builder-value-type-relation-dropdown-matching": "matching",
   "query-builder-value-type-relation-dropdown-without": "without",
-  "query-builder-value-type-relation-dropdown-regardless-of-value": "regardless of value"
+  "query-builder-value-type-relation-dropdown-regardless-of-value": "regardless of value",
+  "query-builder-limit-number-results-description": "Limit the number of results to"
```
Comment (micgro42) [inline public/i18n/en.json:23]: I think the copy doesn't really belong in this PR. Mainly, because I'm not sure that it is already final and so we shouldn't make translators work on things that might be outdated by tomorrow. Let's include that copy as literal text in the component for now

**Review by micgro42 — CHANGES_REQUESTED** [review]

File: public/i18n/en.json (hunk 1 above)
Comment (guergana) [reply public/i18n/en.json:23]: It is not clear in the tickets that this is not the final copy. There is an older version somewhere but this is final as far as I am concerned. Did I miss some discussion about this?

Comment (micgro42) [reply public/i18n/en.json:23]: Yes, this part of the ticket wasn't ready to be worked on in the first place. I think there were still some UX/design questions open about it. At the very least, we need a good idea for how to split the meaning of that single label between input components.

Comment (guergana) [reply public/i18n/en.json:23]: We can discuss it tomorrow in the meeting and leave it here for now maybe

Comment (guergana) [reply public/i18n/en.json:23]: It has been decided that this is the final text for now.

File: src/components/Limit.vue [hunk 2]
```diff
@@ -14,8 +14,8 @@
 			class="querybuilder__limit-input"
 			v-model="textInputValue"
 			:disabled="!checked"
-			:label="$i18n('query-builder-limit-number-results-description')"
-		/> <!-- TODO: decide how the missing label should be handled -->
+			label="Maximum number of results"
```
Comment (guergana) [inline src/components/Limit.vue:17]: at some point we need to also translate this. Also decide the final text

File: src/components/Limit.vue [hunk 3]
```diff
@@ -0,0 +1,87 @@
+<template>
+	<div class="querybuilder__limit">
```
Comment (micgro42) [inline src/components/Limit.vue:2]: this classname isn't exactly ideal BEM, but we have this problem in other places as well. We'll need to do a general sweep of class names to align all of them. But not in this PR.

File: src/store/RootState.ts [hunk 4]
```diff
@@ -5,6 +5,7 @@ export default interface RootState {
 	conditionRows: ConditionRow[];
 	errors: Error[];
 	limit?: number;
```
Comment (micgro42) [inline src/store/RootState.ts]: It makes sense to make that field non-optional right away. We have to touch every test where this is relevant anyway to add the `useLimit` field, so we can add this as well.

**Review by micgro42 — CHANGES_REQUESTED** [review]
Only a small thing to change: making the `limit` field in the store not optional. This should only require another line to some tests that you already touched

After that it can be merged :)

File: src/components/Limit.vue (hunk 3 above)
Comment (guergana) [reply src/components/Limit.vue:2]: oh, i was taking other components as guide. O: yes, makes sense to change them all at once if this is not the best way

**Review by micgro42 — APPROVED** [review]
